js/components/App.js

```diff
@@ -0,0 +1,154 @@
+import Current from './Current';
+import Forecast from './Forecast';
+import Search from './Search';
+import Storage from './Storage';
+import Degree from './Degree';
+import * as utils from './../vendor/utils';
+
+export default class App {
+
+  constructor() {
+    this.state = {
+      current: null,
+      forecast: null,
+      degree: 'M',
+      url_city_name: new URLSearchParams(location.search).get('city') || '',
+    };
+
+    this.request = {
+      API_URL: 'https://api.weatherbit.io/v2.0',
+      API_KEY: '3182fa324b4340ef9cb632451ebb05c1' // e83a8a7ac30d465b93bd8e2bb270bbf7
+    }
+
+    this.bindCallbacks();
+
+    this.search = new Search({onSearch: this.onSearch});
+    this.current = new Current({onStar: this.onStar});
+    this.forecast = new Forecast();
+    this.storage = new Storage({onDropdownItemClick: this.onDropdownItemClick});
+    this.degree = new Degree({onDegreeChange: this.onDegreeChange});
+
+    this.init();
+  }
+
+  bindCallbacks() {
+    this.onSearch = this.onSearch.bind(this);
+    this.onStar = this.onStar.bind(this);
+    this.onDropdownItemClick = this.onDropdownItemClick.bind(this);
+    this.onDegreeChange = this.onDegreeChange.bind(this);
+  }
+
+  init() {
+    const INITIAL_CITY = 'Kiev';
+
+    if (this.state.url_city_name) { // smth present in url ?= -> search it
+      this.search.update({city_name: this.state.url_city_name});
+    } else {
+      if (this.storage.state.history.length !== 0) { // history is not empty -> search by last
+        this.search.update({city_name: JSON.parse(localStorage.getItem('last'))});
+      } else { // history is empty -> search by INITIAL
+        this.search.update({city_name: INITIAL_CITY});
+      }
+    }
+  }
+
+  getWeather(location) {
+    Promise.all([ this.get(`/current?lat=${location.lat}&lon=${location.lng}&units=${this.state.degree}`)
+                    .then(info => info.data[0]),
+                  this.get(`/forecast/daily?lat=${location.lat}&lon=${location.lng}&days=8&units=${this.state.degree}`)
+                    .then(info => info.data) ])
+      .then(data => this.render(data))
+      .then(() => this.addToHistoryStack())
+      .catch(() => alert('Server responded with error. Please, try again later...'));
+  }
+
+  get(query) {
+    const url = `${this.request.API_URL}${query}&key=${this.request.API_KEY}`;
+    return  fetch(url)
+              .then(response => {
+                if (response.ok) {
+                  return response.json();
+                }
+              });
+  }
+
+  updateState(nextState) {
```

Comment:
Please, finish your `Component` class. You can move your method `updateState` there. In such way, all components will have one logic for updating itself.

Reply:
If you need to make some side effects when you updating state, you can create lifecycle methods. Something like this https://github.com/zonzujiro/spa-framework/blob/master/src/framework/Component.js#L38